src/cpp/parse.cpp

```diff
@@ -1238,11 +1236,9 @@ bool Parser::rTemplateDecl(cpp_declarationt &decl)
   {
   case tdk_decl:
     #ifdef DEBUG
-    std::cout << std::string(__indent, ' ') << "BODY: " << body << std::endl;
-    std::cout << std::string(__indent, ' ') << "TEMPLATE_TYPE: "
-              << template_type << std::endl;
+    std::cout << std::string(__indent, ' ') << "BODY\n";
+    std::cout << std::string(__indent, ' ') << "TEMPLATE_TYPE\n";
```

Comment:
It seems `body` and `template_type` have been dropped here - I think you'll need `body.pretty()` and `template_type.pretty()`